server/sonar-web/src/main/js/apps/quality-profiles/utils.js

```diff
@@ -67,6 +62,62 @@ export function createFakeProfile(overrides: {}) {
   };
 }
 
-export function isStagnant(profile: { userUpdatedAt: string }) {
+export function isStagnant(profile: Profile) {
   return moment().diff(moment(profile.userUpdatedAt), 'years') >= 1;
 }
+
+export const getProfilesPath = (organization: ?string) =>
+  organization ? `/organizations/${organization}/quality_profiles` : '/profiles';
+
+export const getProfilesForLanguagePath = (language: string, organization: ?string) => ({
+  pathname: organization ? `/organizations/${organization}/quality_profiles` : '/profiles',
+  query: { language }
+});
+
+export const getProfilePath = (name: string, language: string, organization: ?string) => ({
+  pathname: organization
+    ? `/organizations/${organization}/quality_profiles/show`
+    : '/profiles/show',
+  query: { name, language }
+});
+
+export const getProfileComparePath = (
+  name: string,
+  language: string,
+  organization: ?string,
+  withKey?: string
+) => {
+  const query: Object = { language, name };
+  if (withKey) {
+    Object.assign(query, { withKey });
+  }
+  return {
+    pathname: organization
+      ? `/organizations/${organization}/quality_profiles/compare`
+      : '/profiles/compare',
+    query
+  };
+};
+
+export const getProfileChangelogPath = (
+  name: string,
+  language: string,
+  organization: ?string,
+  filter?: { since?: string, to?: string }
+) => {
+  const query: Object = { language, name };
+  if (filter) {
+    if (filter.since) {
+      Object.assign(query, { since: filter.since });
+    }
+    if (filter.to) {
+      Object.assign(query, { to: filter.to });
+    }
+  }
+  return {
+    pathname: organization
+      ? `/organizations/${organization}/quality_profiles/changelog`
```

Comment:
They could all use the `getProfilesPath` function and then append the specific part after (`/changelog`, `/compare`, `/show`)